File: w1/plus_minus_test.go
package w1

import (
	"strconv"
	"strings"
	"testing"

	"github.com/treejamie/hackerrank-90days/testutils"
)

// Test_plus_minus ensures
func Test_plus_minus(t *testing.T){
	// test cases 
	tcs := []string{
		"tc/1_0.txt",
		"tc/1_1.txt",
		"tc/1_2.txt",
	}

	// now iterate over the test cases
	for _, tc := range tcs {
		// read in the files
		lines, _ :=  testutils.ReadLines(tc)
		// don't want the first line, but want the second
		// and split it up into a slice.
		parts := strings.Fields(lines[1])
		args := make([]int32, len(parts))
		for i, arg := range parts{
			n, _ := strconv.ParseInt(arg, 10, 32)
			args[i] = int32(n)

		}
		
		// we expect an array of strings three items long
		expected := [3]string{
			strings.TrimSpace(lines[3]),
			strings.TrimSpace(lines[4]),
			strings.TrimSpace(lines[5]),
		}

		// call the function
		got := PlusMinus(args)

		// and now compare got with expected
		if got != expected {
			t.Errorf("Test_plus_minus: %q, want %q", got, expected)
		}

	}
}

File: w1/plus_minus_test.py
"""
Tests for plus_minus
"""
from pathlib import Path
from test import testutils
from w1.plus_minus import plus_minus


class PlusMinusTest(testutils.TestCase):
    """
    Ensure everything works as expected
    """

    @classmethod
    def setUpClass(cls):
        """Get all the matching test case files"""
        cls.tcs = sorted(Path("w1/tc").glob("1_*.txt"))

    def test_plus_minus(self):
        """(W1/1): Plus Minus"""
        # iterate over our testcases
        for tc in self.tcs:
            # read in the lines
            lines = list(self.read_lines(Path(tc)))

            # args: line 2 - but it needs splitting up and turning into a list of ints
            args = [int(value) for value in lines[1].split()]

            # expected: lines 4,5,6
            expected = lines[3:6]

            # it returns what we expect
            self.assertEqual(plus_minus(args), expected)
